src/expression.rs

```diff
@@ -38,11 +38,29 @@ pub trait FromTree: Sized {
 }
 
 impl<'a> Tree<'a> {
-    fn from_slice(sl: &'a str) -> Result<(Tree<'a>, &'a str), Error> {
-        Self::from_slice_helper(sl, 0u32)
+    pub fn from_slice(sl: &'a str) -> Result<(Tree<'a>, &'a str), Error> {
```

Comment:
In 73dd23e247dc47fd4d6d50244bc30e27abd3b487, the function signature should be changed in this commit itself.